plugins/backstage-plugin-flux/src/components/FluxRuntimeCard/FluxRuntimeCard.test.tsx

```diff
@@ -0,0 +1,203 @@
+import React from 'react';
+import { renderInTestApp, TestApiProvider } from '@backstage/test-utils';
+import { configApiRef } from '@backstage/core-plugin-api';
+import { ConfigReader } from '@backstage/core-app-api';
+import {
+  KubernetesApi,
+  kubernetesApiRef,
+  KubernetesAuthProvidersApi,
+  kubernetesAuthProvidersApiRef,
+} from '@backstage/plugin-kubernetes';
+import { KubernetesRequestBody } from '@backstage/plugin-kubernetes-common';
+import { FluxRuntimeCard } from './FluxRuntimeCard';
+
+const makeTestFluxController = (
+  name: string,
+  namespace: string,
+  images: string[],
+  clusterName: string,
+  labels: { [name: string]: string },
+) => {
+  return {
+    name,
+    namespace,
+    conditions: [
+      {
+        type: 'Available',
+        status: 'True',
+        reason: 'MinimumReplicasAvailable',
+        message: 'Deployment has minimum availability.',
+        timestamp: '',
+      },
+      {
+        type: 'Progressing',
+        status: 'True',
+        reason: 'NewReplicaSetAvailable',
+        message: 'ReplicaSet has successfully progressed.',
+        timestamp: '',
+      },
+    ],
+    images,
+    suspended: false,
+    clusterName,
+    uid: '4527e05c-eed4-489d-93ae-0cd66ca3277e',
+    labels,
+  };
+};
+
+class StubKubernetesClient implements KubernetesApi {
+  getObjectsByEntity = jest.fn();
+
+  async getClusters(): Promise<{ name: string; authProvider: string }[]> {
+    return [
+      { name: 'mock-cluster-1', authProvider: 'serviceAccount1' },
+      { name: 'mock-cluster-2', authProvider: 'serviceAccount2' },
+    ];
+  }
+
+  getWorkloadsByEntity = jest.fn();
```

Comment:
These two `getWorkloadsByEntity` and `getCustomObjectsByEntity` aren't used?

Reply:
Yup though if I try removing them the stub client throws out this error: 

```
Class 'StubKubernetesClient' incorrectly implements interface 'KubernetesApi'.
  Type 'StubKubernetesClient' is missing the following properties from type 'KubernetesApi': getWorkloadsByEntity, getCustomObjectsByEntity
```